src/astgen

```diff
@@ -137,11 +137,24 @@ sub read_refs {
 
 #----------------------------------------------------------------------
 
-sub open_file {
+sub open_output_file {
     my $filename = shift;
-    my $fh = IO::File->new($filename,"w") or die "%Error: $! $filename,";
-    print $fh '// Generated by astgen // -*- mode: C++; c-file-style: "cc-mode" -*-'."\n";
-    return $fh;
+    if (!defined($filename)) {
+        return \*STDOUT;
+    } else {
+        chmod 0660, $filename if $^O eq "linux" && -e $filename;
```

Comment:
Why is the chmod needed?  This might remove world permission which might be desired.

Reply:
If the output file already exists (as checked in the condition), i.e.: you already run astgen on a previous build, then it was previously made read only by astgen when it output it, so the open for write below would fail without the chmod to add w pemissions. I was under the impression that these are just internal file (and the astgen processed versions of those) used while building Verilator so the permissions should only mater to the dev (who is the user), but not externally.

Reply:
Well same question then about the chmod to make it read only.  Don't think we should be messing with permissions at all.  If you're worried about editing it, your editor probably has a way to make a directory or filename read-only by default, or if you were using emacs we could add the default-to-read-only inside the file itself.

Reply:
No problem. I have removed tha changes to astgen from this PR